nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -117,6 +115,22 @@ in
       '';
     };
 
+    networking.dhcpcd.setHostname = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = ''
+          Whenever to set the machine hostname based on the information
+          received from the DHCP server.
+
+          ::: {.note}
+          The hostname will be changed only if the current one is
+          the empty string, `localhost` or `nixos`.
+
+          Polkit ([](#opt-security.polkit.enable)) is also required.
+          :::
```

Comment:
This text block has 4 space indentation instead of the normal 2?

Reply:
Fixed.